smdebug/core/hook.py

```diff
@@ -653,7 +653,13 @@ def _write_scalars(self):
                     )
             if write_event:
                 if self.writer is None:
-                    self._initialize_writers()
+                    try:
+                        self._initialize_writers()
+                    except NotImplementedError:
+                        # This can happen when the user explicitly calls save_scalars
```

Comment:
not sure if I understand this comment?

Reply:
User Script: 
```
hook = (
...
)
strategy = UndefinedTFStrategy() # strategy not currently supported by smdebug
train() # hook does not save any data. Prints a log msg and turns off
hook.save_scalar("foobar") # Error due to error thrown by line 656
```

Reply:
Is "training method not supported" applicable only to TF or other frameworks too? 

when a strategy is not supported, we set self._hook_supported to False. in general, what happens when the user calls, say, `hook.wrap_optimizer` or `hook.record_tensor_value` when hook is not supported?

Reply:
also, why is this only relevant here and not elsewhere where initialize_writers is called?

can you move this NotImplementedError to the definition of initialize_writers() if it is applicable in other scenarios too?

Reply:
> when a strategy is not supported, we set self._hoo

If `hook._is_not_supported() is True` then we do not save any data.
This check is currently done at the start of all the callbacks.

Data saved by both `hook.wrap_optimizer` and `hook.record_tensor_value` are saved during these callbacks.

save_scalars however writes data outside the callback codepath and hence requires additional checks to ensure that we do not save any data.